paint/PAINT_DrawWindow.h

```diff
@@ -112,5 +91,27 @@ namespace paint
 		void draw() override;
 		void toggleDraw();
 		void clearScreen() const;
+
+	private:
+		// TODO What sort of pointer?
+		Tool* activeTool_;
+		Brush* activeBrush_;
+		//const char* name_;
+		gfx::Colour primaryColour_;
+		gfx::Colour secondaryColour_;
+		std::vector<Coords> clickedPixels_;
+		//SDL_Surface* surface_;
+		SDL_Renderer* renderer_;
+		SDL_Texture* texture_;
+		bool drawToggle_;
+		Coords mouseCoords_;
+		Coords prevMouseCoords_;
+		std::vector<Line> lines_;
+
+		//whenever the active colour is changed this should be updated
+		uint8_t drawRGBA_[4];
+		uint8_t primaryRGBA_[4];
+		uint8_t secondaryRGBA_[4];
```

Comment:
I'm a bit torn by these. They violate a practice called "Single Source of Truth" (SSOT) which is about having a definitive version of information. Here you have two and necessarily have to synchronize them. In circumstances where the colour representation is `float` then they will be less useful, for one thing. The small performance gain in having them is at the cost of reusable engineering, therefore. Plus they clutter the interface. So I suggest they be removed.
**SHOULD**

Reply:
Removed drawRGBA_ -
We want to preserve the primary/secondary colour usage for eg. shapes with outline of primary colour and fill of secondary colour, so I'm removing the drawRGBA_ instead and setting instances where drawRGBA_ is called to instead use primaryRGBA_ or secondaryRGBA_, depending on which one is active.
Required extensive changes throughout DrawWindow and ColourPicker, and redefining how DrawWindow.draw() works.
Changed how the ColourValueTextbox works, previously it linked only to the drawColour, but now it links to both primaryColour and secondaryColour and will change/display the active one at a time.